jaeger-thrift/src/main/java/io/jaegertracing/thrift/internal/senders/UdpSender.java

```diff
@@ -54,14 +57,30 @@ public UdpSender(String host, int port, int maxPacketSize) {
       port = DEFAULT_AGENT_UDP_COMPACT_PORT;
     }
 
-    udpTransport = ThriftUdpTransport.newThriftUdpClient(host, port);
-    agentClient = new Agent.Client(protocolFactory.getProtocol(udpTransport));
+    this.host = host;
+    this.port = port;
+  }
+
+  private Agent.Client getAgentClient() {
+    Agent.Client localRef = this.agentClient;
+    if (localRef == null) {
+      synchronized (this) {
+        localRef = this.agentClient;
+        if (localRef == null) {
+          udpTransport = ThriftUdpTransport.newThriftUdpClient(host, port);
+          localRef = new Agent.Client(protocolFactory.getProtocol(udpTransport));
+          agentClient = localRef;
```

Comment:
```suggestion
          this.agentClient = localRef;
```

To make it easier to determine the context?